hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/ServiceFactory.java

```diff
@@ -102,37 +106,31 @@
      */
     public static final boolean ORDERED_ASYNC_RESPONSES_DEFAULT = true;
 
-    private final boolean isCooperative;
+    private boolean isCooperative = COOPERATIVE_DEFAULT;
 
     // options for async
-    private final int maxPendingCallsPerProcessor;
-    private final boolean orderedAsyncResponses;
+    private int maxPendingCallsPerProcessor = MAX_PENDING_CALLS_DEFAULT;
+    private boolean orderedAsyncResponses = ORDERED_ASYNC_RESPONSES_DEFAULT;
 
     @Nonnull
-    private final FunctionEx<? super Context, ? extends C> createContextFn;
+    private FunctionEx<? super Context, ? extends C> createContextFn;
+
     @Nonnull
-    private final BiFunctionEx<? super Processor.Context, ? super C, ? extends S> createServiceFn;
+    private BiFunctionEx<? super Processor.Context, ? super C, ? extends S> createServiceFn = (ctx, svcContext) -> {
+        throw new IllegalStateException("This ServiceFactory is missing a createServiceFn");
+    };
+
     @Nonnull
-    private final ConsumerEx<? super S> destroyServiceFn;
+    private ConsumerEx<? super S> destroyServiceFn = ConsumerEx.noop();
+
     @Nonnull
-    private final ConsumerEx<? super C> destroyContextFn;
-
-    private ServiceFactory(
-            @Nonnull FunctionEx<? super ProcessorSupplier.Context, ? extends C> createContextFn,
-            @Nonnull BiFunctionEx<? super Processor.Context, ? super C, ? extends S> createServiceFn,
-            @Nonnull ConsumerEx<? super S> destroyServiceFn,
-            @Nonnull ConsumerEx<? super C> destroyContextFn,
-            boolean isCooperative,
-            int maxPendingCallsPerProcessor,
-            boolean orderedAsyncResponses
-    ) {
+    private ConsumerEx<? super C> destroyContextFn = ConsumerEx.noop();
+
+    @Nonnull
+    private Set<File> filesToAttach = new HashSet<>();
```

Comment:
this would get serialized

Reply:
Fixed by nulling out in `attachedFiles` call.